src/Microsoft.Framework.Configuration/ConfigurationKeyComparer.cs

```diff
@@ -8,14 +8,12 @@ namespace Microsoft.Framework.Configuration
 {
     public class ConfigurationKeyComparer : IComparer<string>
     {
-        private const char Separator = ':';
-
         public static ConfigurationKeyComparer Instance { get; } = new ConfigurationKeyComparer();
 
         public int Compare(string x, string y)
         {
-            var xParts = x?.Split(Separator) ?? new string[0];
-            var yParts = y?.Split(Separator) ?? new string[0];
+            var xParts = x?.Split(new string[]{ Constants.KeyDelimiter }, StringSplitOptions.RemoveEmptyEntries) ?? new string[0];
```

Comment:
If you want to get a little fancier, you could declare a `private static readonly string[] KeyDelimiterArray = ...` so it's not creating a new array every time this method is called.